sympy/matrices/expressions/matmul.py

```diff
@@ -120,6 +120,37 @@ def doit(self, **kwargs):
             args = self.args
         return canonicalize(MatMul(*args))
 
+
+    def expand(self, *args, **kwargs):
+        #expand MatMul
+        # e.g.
+        # A *(B + C) = A*B + A*C
+        # (B+C)*A = B*A + B*C
+        # expand((A+D)*(B + C), deep=False) = (A+D) *B + (A+D)*C
```

Comment:
I don't believe this matches the response when dealing with Expr...deep refers to traversing arguments of functions.